### PR TITLE
[Tech] update get user playstreak

### PR DESCRIPTION
**Pull request by BrettCleary**
# Summary

Let's export the query options separately so that they can also be used with `useQueries` in the consuming code without creating two duplicate `queryFn`'s that could return different data than the other.

This is need for a refactor in https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/pull/1124

### CODE REVIEW

File: src/hooks/useGetUserPlayStreak.ts [hunk 1]
```diff
@@ -2,25 +2,33 @@ import { getGetUserPlayStreakQueryKey } from '@/helpers/getQueryKeys'
...
+  return {
     queryKey,
     queryFn: async () => {
       if (questId === null) {
         return null
       }
       const response = await getUserPlayStreak(questId)
       if (!response) return null
-      return response
+      return { questId: questId, userPlayStreak: response }
```
Comment (BrettCleary) [inline src/hooks/useGetUserPlayStreak.ts:18]: the user playstreak response does not have the quest id, but I need it for the `useQueries` call in https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/pull/1124/files#diff-191d2dad24a032e65214284f9b14a96e4f1445c55a177f60c677a169185db5c3 `src/frontend/hooks/useGetQuestStates.ts`

File: src/hooks/useGetQuest.ts [hunk 2]
```diff
@@ -21,6 +20,15 @@ export function useGetQuest(
...
     enabled: questId !== null
   })
+}
+
+export function useGetQuest(
+  questId: number | null,
+  getQuest: (questId: number) => Promise<Quest>
+) {
+  const queryClient = useQueryClient()
+  const queryKey = getGetQuestQueryKey(questId)
```
Comment (eliobricenov) [inline src/hooks/useGetQuest.ts:30]: same comment from the other hook

File: src/hooks/useGetUserPlayStreak.ts [hunk 3]
```diff
@@ -1,26 +1,36 @@
...
     enabled: questId !== null
   })
+}
+
+export function useGetUserPlayStreak(
+  questId: number | null,
+  getUserPlayStreak: (questId: number) => Promise<UserPlayStreak>
+) {
+  const queryClient = useQueryClient()
+  const queryKey = getGetUserPlayStreakQueryKey(questId)
```
Comment (eliobricenov) [inline src/hooks/useGetUserPlayStreak.ts:30]: Why are we returning the whole query under data? Also, I think we can get the queryKey from the options directly.

```ts
const queryOptions = getUserPlaystreakQueryOptions(questId, getUserPlayStreak);
const query = useQuery(queryOptions)

return {
    ...query,
    isLoading: query.isLoading || query.isFetching,
    invalidateQuery: async () => queryClient.invalidateQueries({ queryKey: queryOptions.queryKey })
  }
```

Comment (BrettCleary) [reply src/hooks/useGetUserPlayStreak.ts]: This is the pattern we use elsewhere in the client. I think we should keep the query props separated from the top level of this object so we don't overwrite them like we would be doing here with `isLoading`. It is a relatively minor stylistic difference.

Just updated to get the query key from the options. Good call.

File: src/hooks/useGetQuest.ts (hunk 2 above)
Comment (BrettCleary) [reply src/hooks/useGetQuest.ts]: Just updated to get the query key from the options.

**Review by eliobricenov — APPROVED** [review]

**Pull request merged**